src/tools/inner_node.rs

```diff
@@ -118,8 +122,24 @@ impl InnerNode {
 
     /// Connects to the target address.
     pub async fn connect(&self, target: SocketAddr) -> io::Result<()> {
-        self.node.connect(target).await?;
-        Ok(())
+        self.node.connect(target).await
+    }
+
+    /// Connects to the target address using outgoing_addr as source address/port.
+    /// Sets `reuse_addr` and `reuse_port` on the outgoing socket.
+    pub async fn connect_with_outgoing_addr(
+        &self,
+        outgoing_addr: SocketAddr,
+        target: SocketAddr,
+    ) -> io::Result<()> {
+        let socket = match target {
+            SocketAddr::V4(_) => TcpSocket::new_v4()?,
+            SocketAddr::V6(_) => TcpSocket::new_v6()?,
+        };
+        socket.bind(outgoing_addr)?;
+        socket.set_reuseaddr(true)?;
+        socket.set_reuseport(true)?;
+        self.node.connect_using_socket(target, socket).await
```

Comment:
If we put this here, we cannot prepare the socket before starting the critical section of the performance test.
Maybe we can leave it out and give the user more control here just for that case above. Otherwise, we'll suffer a performance hit.